sky/provision/azure/config.py

```diff
@@ -142,12 +144,15 @@ def bootstrap_instances(
     cluster_id, nsg_name = get_cluster_id_and_nsg_name(
         resource_group=provider_config['resource_group'],
         cluster_name_on_cloud=cluster_name_on_cloud)
+
+    # subnet_mask is generated only for new subnets
     subnet_mask = provider_config.get('subnet_mask')
-    if subnet_mask is None:
-        # choose a random subnet, skipping most common value of 0
-        random.seed(cluster_id)
-        subnet_mask = f'10.{random.randint(1, 254)}.0.0/16'
-    logger.info(f'Using subnet mask: {subnet_mask}')
+    # choose a random subnet, skipping most common value of 0
+    random.seed(cluster_id)
+    subnet_mask = f'10.{random.randint(1, 254)}.0.0/16'
+    if subnet_id == '':
```

Comment:
why do we change this from subnet_mask is missing, to subnet_id is missing? what is the relation between these two variables?